### PR TITLE
Correction before session 2021

### PR DESCRIPTION
**Pull request by madvid**
* multiple typo and errors correction
* new exercises
* maths coherence in dimension
* more precision in docstrings
*  etc

### CODE REVIEW

File: module05/ex00/ex00.md [hunk 1]
```diff
@@ -11,19 +11,28 @@
  **AI Classics:**   
 *These exercises are key assignments from the last bootcamp. If you haven't completed them yet, you should finish them first before you continue with today's exercises.*
 
-You will provide a testing file to prove that your class works as expected.  
-You will have to create a helpful class, with more options and providing enhanced ease of use for the user.
-
+## Intructions:
```
Comment (ppeigne) [inline module05/ex00/ex00.md:14]: Instructions instead ?

File: module05/ex00/ex00.md [hunk 2]
```diff
@@ -11,19 +11,28 @@
...
+You will also provide a testing file `test.py` to demonstrate your class works as expected.
+
+## Examples:
 ```py
 # Column vector of dimensions n * 1
-v1 = Vector([[0.0], [1.0], [2.0], [3.0]])
-v1 * 5
-# Output
-Vector([[0.0], [5.0], [10.0], [15.0]])
+>> v1 = Vector([[0.0], [1.0], [2.0], [3.0]])
```
Comment (ppeigne) [inline module05/ex00/ex00.md:32]: Le modèle classique d'exemples ne prends pas de chevrons et ne fait pas appel à print pour présenter les résultats attendus (cf tous les autres exemples ci après)

Ex: 
```
# Column vector of dimensions n * 1
v1 = Vector([[0.0], [1.0], [2.0], [3.0]])
v2 = Vector([[2.0], [1.5], [2.25], [4.0]])
v1.dot(v2)
# Output
18
```

File: module05/ex02/ex02.md [hunk 3]
```diff
@@ -11,32 +11,33 @@
 **AI Classics:**   
 *These exercises are key assignments from the last bootcamp. If you haven't completed them yet, you should finish them first before you continue with today's exercises.*
 
+## Instructions:
 Create a class named `TinyStatistician` which implements the following methods.   
-All methods take in an list and return a new modified one.
+All methods take a `list` or a numpy.`ndarray` as parameter.
```
Comment (ppeigne) [inline module05/ex02/ex02.md:16]: Numpy est interdite comme librairie sur cet exo, est-ce normal d'accepter les nd.array en input ?

File: module05/ex02/ex02.md [hunk 4]
```diff
@@ -11,32 +11,33 @@
...
 
-* `mean(x)` : computes the mean of a given non-empty list `x`, using a for-loop and returns the mean as a float, otherwise None if `x` is an empty list or a non list object. This method should not raise any Exception.
+* `mean(x)` : computes the mean of a given non-empty list or array `x`, using a for-loop. The method returns the mean as a float, otherwise None if `x` is an empty list or array, or a non expected type object. This method should not raise any Exception.
```
Comment (ppeigne) [inline module05/ex02/ex02.md:18]: Numpy forbidden ?

File: module05/ex02/ex02.md [hunk 5]
```diff
@@ -11,32 +11,33 @@
...
 
 Given a vector $x$ of dimension m * 1, the mathematical formula of its mean is:
 
 $$
 \mu = \frac{\sum_{i = 1}^{m}{x_i}}{m}
 $$
 
-* `median(x)` : computes the median, also called the 50th percentile, of a given non-empty list `x`, using a for-loop and returns the median as a float, otherwise None if `x` is an empty list or a non list object. This method should not raise any Exception.
+* `median(x)` : computes the median, which is also the 50th percentile, of a given non-empty list or array `x` . The method returns the median as a float, otherwise None if `x` is an empty list or array or a non expected type object. This method should not raise any Exception.
```
Comment (ppeigne) [inline module05/ex02/ex02.md:26]: Numpy forbidden ?

File: module05/ex02/ex02.md [hunk 6]
```diff
@@ -11,32 +11,33 @@
...
 
-* `quartile(x)` : computes the 1st and 3rd quartiles, also called the 25th percentile and the 75th percentile, of a given non-empty list `x`, and returns the quartiles as a list of 2 floats, otherwise None if `x` is an empty list or a non list object. This method should not raise any Exception.
+* `quartile(x)` : computes the 1st and 3rd quartiles, also called the 25th percentile and the 75th percentile, of a given non-empty list or array `x`. The method returns the quartiles as a list of 2 floats, otherwise None if `x` is an empty list or array or a non expected type object. This method should not raise any Exception.
```
Comment (ppeigne) [inline module05/ex02/ex02.md:28]: Numpy forbidden ?

File: module05/ex02/ex02.md [hunk 7]
```diff
@@ -11,32 +11,33 @@
...
 
-* `percentile(x, percentile)` : computes the expected percentile of a given non-empty list `x`, and returns the percentile as a float, otherwise None if `x` is an empty list or a non list object. The second parameter is the wished percentile. This method should not raise any Exception.
+* `percentile(x, percentile)` : computes the expected percentile of a given non-empty list or array `x`. The method returns the percentile as a float, otherwise None if `x` is an empty list or array or a non expected type object. The second parameter is the wished percentile. This method should not raise any Exception.
```
Comment (ppeigne) [inline module05/ex02/ex02.md:30]: Numpy forbidden ?

File: module05/ex02/ex02.md [hunk 8]
```diff
@@ -11,32 +11,33 @@
...
 
-* `var(x)` : computes the variance of a given non-empty list `x` and returns the variance as a float, otherwise None if `x` is an empty list or a non list object. This method should not raise any Exception.
+* `var(x)` : computes the variance of a given non-empty list or array `x`. The method returns the variance as a float, otherwise None if `x` is an empty list or array or a non expected type object. This method should not raise any Exception.
```
Comment (ppeigne) [inline module05/ex02/ex02.md:32]: Numpy forbidden ?

File: module05/ex02/ex02.md [hunk 9]
```diff
@@ -11,32 +11,33 @@
...
 
 Given a vector $x$ of dimension m * 1, the mathematical formula of its variance is:
 
 $$
 \sigma^2 = \frac{\sum_{i = 1}^{m}{(x_i - \mu)^2}}{m} = \frac{\sum_{i = 1}^{m}{[x_i - (\frac{1}{m}\sum_{j = 1}^{m}{x_j}})]^2}{m}
 $$
 
-* `std(x)` : computes the standard deviation of a given non-empty list `x`, and returns the standard deviation as a float, otherwise None if `x` is an empty list or a non list object. This method should not raise any Exception.
+* `std(x)` : computes the standard deviation of a given non-empty list or array `x`. The method returns the standard deviation as a float, otherwise None if `x` is an empty list or array or a non expected type object. This method should not raise any Exception.
```
Comment (ppeigne) [inline module05/ex02/ex02.md:40]: Numpy forbidden ?

File: module05/ex03/ex03.md [hunk 10]
```diff
@@ -33,13 +33,14 @@ def simple_predict(x, theta):
     Args:
       x: has to be an numpy.ndarray, a vector of dimension m * 1.
       theta: has to be an numpy.ndarray, a vector of dimension 2 * 1.
-    Returns:
+    Return:
```
Comment (ppeigne) [inline module05/ex03/ex03.md:36]: Should be returns instead (3rd pers.), same as RaiseS below.

File: module06/ex03/ex03.md [hunk 11]
```diff
@@ -22,13 +23,13 @@ $$
...
 Where:  
-- $\nabla(J)$ is the gradient vector of size 2 * 1, (this strange symbol : $\nabla$ is called nabla)
-- $x$ is a vector of dimension m * 1
-- $y$ is a vector of dimension m * 1
-- $x^{(i)}$ is the $i^{th}$ component of vector $x$
-- $y^{(i)}$ is the $i^{th}$ component of vector $y$
-- $\nabla(J)_j$ is the $j^{th}$ component of $\nabla(J)$
-- $h_{\theta}(x^{(i)})$ corresponds to the model's prediction of $y^{(i)}$
+- $\nabla(J)$ is the gradient vector of size $2 \times 1$, (this strange symbol : $\nabla$ is called nabla)
+- $x$ is a vector of dimension $m * 1$,
```
Comment (ppeigne) [inline module06/ex03/ex03.md:27]: use "\times" instead of "*" ?

File: module08/ex12/ex12.md [hunk 12]
```diff
@@ -2,13 +2,13 @@
...
-|   Turn-in directory :   |  ex12                   |
+|   Turn-in directory :   |  ex12/                  |
 |   Files to turn in :    |  confusion_matrix.py    |
 |   Forbidden functions : |  None                   |
 |   Remarks :             |  n/a                    |
 
-## Objectives:
-
+## Objective:
+Manipulation oof confusion matrix concept.
```
Comment (ppeigne) [inline module08/ex12/ex12.md:11]: of

**Review by ppeigne — COMMENTED** [review]
Great job! Few typos to remove and its perfect

File: module06/ex03/ex03.md (hunk 11 above)
Comment (ppeigne) [reply module06/ex03/ex03.md:27]: Actually should be just dimension $m$

**Review by ppeigne — DISMISSED** [review]
All good :)

File: module05/ex00/ex00.md (hunk 2 above)
Comment (madvid) [reply module05/ex00/ex00.md]: J'ai pas compris, j'ai oublié de retirer des chevrons ?

**Review by qfeuilla — CHANGES_REQUESTED** [review]

File: module08/ex03/ex03.md [hunk 13]
```diff
@@ -2,21 +2,22 @@
...
 
 $$
 \text{sigmoid}(x) = \cfrac{1} {1 + e^{-x}}
 $$
 
 Where:  
-* $x$ is a scalar or a vector
-* $e$ is a mathematical constant, named Euler's number
+* $x$ is a scalar or a vector,
+* $e$ is the contracted form for exponential function. It is also a mathematical constant, named Euler's number.
```
[the remark this answers is not in the document]
Comment (qfeuilla) [reply module08/ex03/ex03.md:20]: forget it, contracted is right

File: module05/ex00/ex00.md [hunk 14]
```diff
@@ -11,19 +11,28 @@
...
-- Columns vectors are represented as list of lists of floats.
+- Columns vectors are represented as list of lists of a single floats.
 - Row vectors are represented as lists of floats.
 
+The class should also have 2 attributes:  
+* `values`: a list (or a list of lists) of floats,
+* `shape`: dimension of the vector.
+
+Finally you have to implement 2 methods:
+* `dot()`: produce a dot product with the vector given as parameter (should have the of same shape),
```
[the remark this answers is not in the document]
Comment (ppeigne) [reply module05/ex00/ex00.md:24]: I sugest: "(both vectors should have the same shape)"

File: module07/ex12/ex12.md [hunk 15]
```diff
@@ -1,21 +1,20 @@
...
+## Objective:  
+It's reflection time!
 
 ## Instructions:
-* For each model you built in the last exercise, plot its hypothesis function $h(\theta)$ on top of a scatter plot of the original data points $(x,y)$. 
+Based on the plots you made in the previous exercise for each model you built, answer the following questions (in answers.txt):
 
-* Answer the following questions: 
-  1. From a purely intuitive point of view, which hypothesis (i.e. which polynomial degree) seems to best represent the relationship between $y$ and $x$?  
+1. From a purely intuitive point of view, which hypothesis (i.e. which polynomial degree) seems to **better represent the relationship** between $y$ and $x$?
```
[the remark this answers is not in the document]
Comment (ppeigne) [reply module07/ex12/ex12.md:16]: Point of view is OK in english (standpoint is a synonym).

File: module09/ex00/ex00.md [hunk 16]
```diff
@@ -1,20 +1,20 @@
...
 *These exercises are key assignments from the previous module. If you haven't completed them yet, you should finish them first before you continue with today's exercises.*
 
-## Objectives:
+## Objective:
 The time to use everything you built so far has come! Demonstrate your knowledge by implementing a logistic regression classifier using the gradient descent algorithm.
-You must have seen the power of `NumPy` for vectorized operations. Well let's make something more concrete with that.
+You must have seen the power of `numpy` for vectorized operations. Well let's make something more concrete with that.
 
-You may have had a look at Scikit-Learn's implementation of logistic regression and noticed that the `sklearn.linear_model.LogisticRegression` class offers a lot of options.
+You may have to take a look at Scikit-Learn's implementation of logistic regression and noticed that the `sklearn.linear_model.LogisticRegression` class offers a lot of options.
```
[the remark this answers is not in the document]
Comment (ppeigne) [reply module09/ex00/ex00.md:17]: If you had taken a look at ... you should have noticed that ?

File: module08/ex12/ex12.md [hunk 17]
```diff
@@ -123,4 +125,4 @@ confusion_matrix_(y_true, y_hat, labels=['bird', 'dog'], df_option=True)
  dog          0    2
 ```
 
-**N.B:** If you fail this exercise on your first attempt, Norminet will curse you forever. Yeah, you'd better do it right or you are in trouble my friend, big trouble !
+**N.B:** If you fail this exercise on your first attempt, Norminet will curse you forever. Yeah, you'd better do it right or you are in trouble my friend, big trouble!
```
[the remark this answers is not in the document]
Comment (madvid) [reply module08/ex12/ex12.md:128]: For the moment, it is not important
